### ISSUE TITLE
Obtaining correct course codes?

### ISSUE THREAD

**Issue by pwinslow** [coursera-dl/coursera-dl#420]
I'm currently taking the econometrics course on coursera. The course info page is here https://www.coursera.org/learn/erasmus-econometrics/. Your python script seems to need current class name but where do I find this quantity? I've tried putting in erasmus-econometrics, erasmus-econometrics-001, erasmus-econometrics-00X, erasmus, econometrics, ... It seems clear that there must be a code for the class that is not necessarily given from coursera. I'm on MAC OSX 10.10, installed coursera-dl via the virtual python environment as on https://github.com/coursera-dl/coursera-dl. The error message I keep getting is below. I think the main problem is that I have the wrong course code, could you please point me to where I could find the class code? Also, if you think there's a different issue at work, could you please let me know what you think?

Thanks very much in advance for any advice/help you may have.

Best,
Peter

(my-coursera)Peters-MacBook-Pro:coursera pwinslow$ ./coursera-dl -u <MYUSERNAME> -p <MYPASSWORD> erasmus-econometrics
Downloading class: erasmus-econometrics
Starting new HTTPS connection (1): class.coursera.org
/Users/pwinslow/CourseraStuff/my-coursera/lib/python2.7/site-packages/requests/packages/urllib3/util/ssl_.py:100: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
404 Client Error: Not Found for url: https://class.coursera.org/erasmus-econometrics
Starting new HTTPS connection (1): www.coursera.org
/Users/pwinslow/CourseraStuff/my-coursera/lib/python2.7/site-packages/requests/packages/urllib3/util/ssl_.py:100: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
Traceback (most recent call last):
  File "./coursera-dl", line 6, in <module>
    coursera_dl.main()
  File "/Users/pwinslow/CourseraStuff/my-coursera/coursera/coursera/coursera_dl.py", line 1111, in main
    if download_class(args, class_name):
  File "/Users/pwinslow/CourseraStuff/my-coursera/coursera/coursera/coursera_dl.py", line 1089, in download_class
    return download_on_demand_class(args, class_name)
  File "/Users/pwinslow/CourseraStuff/my-coursera/coursera/coursera/coursera_dl.py", line 1028, in download_on_demand_class
    login(session, args.username, args.password)
  File "/Users/pwinslow/CourseraStuff/my-coursera/coursera/coursera/cookies.py", line 121, in login
    headers=headers, allow_redirects=False)
  File "/Users/pwinslow/CourseraStuff/my-coursera/lib/python2.7/site-packages/requests/sessions.py", line 511, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/Users/pwinslow/CourseraStuff/my-coursera/lib/python2.7/site-packages/requests/sessions.py", line 468, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/pwinslow/CourseraStuff/my-coursera/lib/python2.7/site-packages/requests/sessions.py", line 576, in send
    r = adapter.send(request, **kwargs)
  File "/Users/pwinslow/CourseraStuff/my-coursera/lib/python2.7/site-packages/requests/adapters.py", line 433, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: [Errno 1] _ssl.c:507: error:14094410:SSL routines:SSL3_READ_BYTES:sslv3 alert handshake failure

**Comment by balta2ar** [comment]
Correct course name is `erasmus-econometrics`.

How exactly did you install virtualenv and coursera-dl? Take a look at this comment, it might help: https://github.com/coursera-dl/coursera-dl/issues/416#issuecomment-150902837

**Comment by lcguida** [comment]
Well, for what I've seen, Coursera is applying new design to some courses. The new courses now have a leading `/learn/` path.

For example, for this course: https://www.coursera.org/learn/learning-how-to-learn, I've tried the following command:

```shell
[$] coursera-dl -n  learning-how-to-learn
....
404 Client Error: Not Found for url: https://class.coursera.org/learning-how-to-learn
Could not find class: learning-how-to-learn
```

And he can't find the correct course path. But, some older courses still have their path like `https://class.coursera.org/<course_name>`.

**Comment by balta2ar** [comment]
@lcguida Try using the latest version from github.

**Comment by ryanschneider** [comment]
@balta2ar FWIW, the latest master on github just worked for me and the `machine-learning` class (which is at `/learn/machine-learning`).  Previously I tried `ml` (since class.coursera.org/ml redirected to /learn/machine-learning) with the version installed by pip but I just got the `Found 0 sections and 0 lectures on this page` result.

Thanks for the tool, I'm happy to have the videos available offline now so I can quickly refer to them at work. :)

**Comment by kenng** [comment]
latest version is not working yet. 
it seems what @lcguida suggests might be a solution, when I try to put url 
class.coursera.org/learn/machine-learning, it will redirect to https://www.coursera.org/learn/machine-learning while https://class.coursera.org/machine-learning gives 

`Sorry, the class you were looking for cannot be found. Please check your URL and try again.`

**Comment by ryanschneider** [comment]
@kenng are you using `master` from github, or the latest version available via pip?  It doesn't look like a new version has been tagged yet.  FWIW, I used commit 45bdcaca4df863ce06e77720aa9fecd6e3886437.  The following command worked for me:

```bash
coursera-dl -u $USERNAME machine-learning --debug
```

I just verified it's still working for me as well.  Note the `--debug` is unnecessary but might help if you hit an issue.
